### ISSUE TITLE
Slow transfer speed

### ISSUE THREAD

**Issue by heixiaoma** [quic-go/quic-go#4533]
I'm not sure if the official team has conducted any transmission stress testing. Currently, in my production environment, the server has a bandwidth of 300Mbps, and the local upload bandwidth is around 100Mbps. In a Windows environment, I have observed upload speeds of (4-6) MB/s. On the same server, using the quiche library(https://github.com/cloudflare/quiche), I can achieve speeds of up to 10+MB/s. However, when using GoQUIC, the transmission performance is halved. I'm unsure if the official team has any restrictions on default configurations in this area.
﻿
﻿

**Comment by heixiaoma** [comment]
quic-go

**Comment by heixiaoma** [comment]
I believe it is a problem that should be thoroughly discussed rather than simply being closed or ignored, including the UDP buffer settings. These optimizations have already been configured, but the speed is indeed not ideal

**Comment by marten-seemann** [comment]
The maintainers might be more willing to engage with your alleged problem if they could see that you put at least a tiny amount of effort in creating the issue.

Getting the name of the library right seems like a pretty low bar.
Insinuating that we didn't do any performance benchmark doesn't earn you any sympathy either (and is completely wrong).
Most importantly, just complaining "quic-go is slow", without any more information on hardware / software / network configuration and without any attached logs is a pretty meaningless statement.

**Comment by heixiaoma** [comment]
> The maintainers might be more willing to engage with your alleged problem if they could see that you put at least a tiny amount of effort in creating the issue.
> 
> Getting the name of the library right seems like a pretty low bar. Insinuating that we didn't do any performance benchmark doesn't earn you any sympathy either (and is completely wrong). Most importantly, just complaining "quic-go is slow", without any more information on hardware / software / network configuration and without any attached logs is a pretty meaningless statement.

Yes, it's possible that my description of the problem is quite direct, or perhaps the translation software has made the communication seem rather brusque. I have reviewed past issues and there have been discussions on similar topics. Currently, quic-go is the only outstanding networking library in Go. Raising this issue is at least a step towards making this library even better.

**Comment by heixiaoma** [comment]
The client currently connects to the server, and the server creates a new data stream. Subsequently, the client sends a file to the server via the new stream. I have recorded the local qlog here, but I do not know how to analyze the log. I hope the author can help to review this log. Currently, I have uploaded a 500MB file, resulting in a 312MB log. The log file is stored on Google Drive. I would appreciate it if the author could assist in analyzing the reasons for this or confirm if there are no issues.

https://drive.google.com/file/d/1rTvseqZlg_mC8-QvlhzroMzaZQmw8GOA/view?usp=sharing

**Comment by heixiaoma** [comment]
![Uploading image.png…]()
qvis report photo

**Comment by heixiaoma** [comment]
https://drive.google.com/file/d/1VwEQ5m1Tk-xMqcFQ_8IGArZ6N5hy-Mqd/view?usp=sharing

This qlog file contains logs generated using the netty-quic library, which is based on the quiche library. The author can refer to and compare the performance of the two by testing them. The comparison shows that quiche is 1-3 MB/s faster than quic-go, although quic-go is also very stable. Hopefully, this information can help the author with their analysis and bring some joy!

**Comment by heixiaoma** [comment]
> https://drive.google.com/file/d/1VwEQ5m1Tk-xMqcFQ_8IGArZ6N5hy-Mqd/view?usp=sharing
> 
> This qlog file contains logs generated using the netty-quic library, which is based on the quiche library. The author can refer to and compare the performance of the two by testing them. The comparison shows that quiche is 1-3 MB/s faster than quic-go, although quic-go is also very stable. Hopefully, this information can help the author with their analysis and bring some joy!

![image](https://github.com/quic-go/quic-go/assets/10923974/874d8cf2-2b09-44bf-ae4f-3ce464ee00e0)

**Comment by heixiaoma** [comment]
Thank you for the author for reopening the issue. Currently, I observed that the congestion window sizes in the two screenshots are somewhat different. At the same time, in the document, I see future plans (https://quic-go.net/docs/quic/congestion-control). Will the performance be better in later versions? If so, perhaps this topic can indeed be closed.
